src/app/services/selector_matcher.js

```diff
@@ -125,14 +127,33 @@ function getNodesByPath(elements, path) {
     return nodes;
 }
 
+function getNodesByFile(elements, path) {
+    var nodes = [];
+
+    _.each(elements, function(node) {
+        var path_parts = node.data.original_file_path.split("/");
+        var fname = _.last(path_parts);
+        if (fname == path) {
+            nodes.push(node.data);
+        }
+    })
+    return nodes;
+}
```

Comment:
I don't think lodash is necessary here, and we could probably leverage `reduce` as well:
```suggestion
function getNodesByFile(elements, path) {
    return elements.reduce((acc, node) => {
      const pathParts = node.data.original_file_path.split("/");
      const fname = pathParts[pathParts.length - 1];
      if (fname === path) {
          acc.push(node.data);
      }
      return acc;
    });
}
```